.editorconfig

```diff
@@ -0,0 +1,15 @@
+# EditorConfig <http://EditorConfig.org>
```

Comment:
Is there a particular reason we are using EditorConfig? The VS Code extensions appears abandoned, there hasn't been an update in over 2 years. Some of these configurations are better left to other tools such as clang-format for C++ files.

I suggest we use Prettier, which provides the same general functionality as EditorConfig but is much more active.

Reply:
We can change to using what every tool for formatting if there is something better. I would suggest we stick with EditorConfig at least for this PR so we have something since we are already using it in other carma apt repos like carma-time-library. We can put in an issue to update this to a different tool and possibly include the configurations in a common place, like the carma-builds images. Just a suggestion